src/types/common.d.ts

```diff
@@ -168,13 +166,10 @@ export type WhereConditions<T extends typeof AbstractBone> = {
   [key in '$and' | '$or']?: WhereConditions<T>[];
 }
 
-export type Values<T extends typeof AbstractBone> = {
-  [Property in keyof Extract<InstanceType<T>, Literal>]?: Literal;
-}
+// https://stackoverflow.com/a/68077021/179691
+export type PickTypeKeys<Obj, Type, T extends keyof Obj = keyof Obj> = ({ [P in keyof Obj]: Obj[P] extends Type ? P : never })[T];
 
-export type InstanceValues<T> = {
-  [Property in keyof Extract<T, Literal>]?: Extract<T, Literal>[Property]
```

Comment:
`Extract<T, Literal>` extracts type Literal from union T, which does not match what we need here.